mongo/options/clientoptions.go

```diff
@@ -121,6 +121,7 @@ type ClientOptions struct {
 	ServerAPIOptions         *ServerAPIOptions
 	ServerSelectionTimeout   *time.Duration
 	SocketTimeout            *time.Duration
+	SRVMaxHosts              *int
```

Comment:
Let's add a test for this in `clientoptions_test.go` - we didn't catch this in the SRV polling tests because they construct a `Topology` directly, which is unfortunate. I'm hoping [GODRIVER-1760](https://jira.mongodb.org/browse/GODRIVER-1760) will help us clean up these abstractions and catch such issues quicker.

Reply:
Sounds good; done.